.github/workflows/build-and-test.yml

```diff
@@ -187,20 +185,20 @@ jobs:
             - name: Fetch git tags
               run: git fetch origin +refs/tags/*:refs/tags/*
 
-            - name: Setup conda environment
-              uses: conda-incubator/setup-miniconda@v2
+            - name: Set up conda environment
+              uses: ./.github/actions/setup_env
               with:
-                  activate-environment: source-env
-                  auto-update-conda: true
-                  python-version: ${{ matrix.python-version }}
-                  channels: conda-forge
+                  cache-id: py${{ matrix.python-version }}-${{ matrix.os }}-test-sdist
+              id: condacache
 
             - name: Install dependencies
+              if: steps.condacache.outputs.cache-hit != 'true'
               shell: bash -l {0}
               run: |
+                  conda install python=${{ matrix.python-version }}
                   conda install nomkl # make sure numpy is w/out MKL
                   conda upgrade -y pip setuptools
-                  conda install twine build
+                  conda install twine build setuptools_scm
```

Comment:
Need `setuptools_scm` for versioning to work right.

Reply:
Can you comment more on this? Is this related to the updates of using` python -m setuptools_scm`?

Reply:
In another PR I'd removed a bunch of extra dependencies that were being installed but not used in some jobs. Turns out that `setuptools_scm` actually is needed and the versioning fails silently without it (the version will be an empty string). So just correcting that mistake.